src/models/app_data_json_format.rs

```diff
@@ -11,19 +12,32 @@ pub struct Referrer {
     pub version: String,
 }
 
+#[serde_as]
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Quote {
+    #[serde(with = "u256_decimal")]
+    pub sell_amount: U256,
+    #[serde(with = "u256_decimal")]
+    pub buy_amount: U256,
+    pub version: String,
+}
+
 #[serde_as]
 #[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash, Default)]
 pub struct Metadata {
+    // we make all of the field optional, in order to be compatible with all versions
```

Comment:
I think the long term correct solution (which would allow backwards-compatability-breaking changes to existing fields) would be something like:

```rust
#[derive(Deserialize, Serialize)]
#[serde(untagged)]
pub enum Metadata {
    V3(MetadataV3),
    V2(MetadataV2),
    V1(MetadataV1),
}
```

I think this approach is 100% fine for now (so not suggesting you change it just yet), just wanted to share the idea as a work-around to when we have backwards-compatability issues on specific fields (like `environment` becomes a `struct` instead of a `String`).